components/brave_extension/extension/brave_extension/background/reducers/shieldsPanelReducer.ts

```diff
@@ -340,6 +347,45 @@ export default function shieldsPanelReducer (
       })
       break
     }
+    case shieldsPanelTypes.GENERATE_CLASS_ID_STYLESHEET: {
+      const tabData = state.tabs[action.tabId]
+      if (!tabData) {
+        console.error('Active tab not found')
+        break
+      }
+      const exceptions = tabData.cosmeticFilters.ruleExceptions
+
+      injectClassIdStylesheet(action.tabId, action.classes, action.ids, exceptions)
```

Comment:
It could be a good idea to wrap this side-effect in a `setTimeout(fn, 0)` since that function could call another redux function immediately. I know it's inside a callback, but for all we know that callback function could get called immediately, before the current redux action is completed.